### PR TITLE
[CC-31] Get translation messages into Transifex

### PR DESCRIPTION
**Pull request by dpoirier**
## Description
Part 1 of CC-31 is to use the Transifex API to create resources
in Transifex and upload the translation messages (source/english
messages and all the translated messages we have).
## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -132,3 +140,61 @@ Deployment with Dokku
...
+First, note that translation is done in two different ways. Most things use the built-in
+Django translation support. But the translation of the actual legal text of the licenses
+is handled separately, and that's what this section is about.
+
+Second note: the initial implementation focuses on the 4.0 by-*
+licenses. Others will be added as time allows.
+
+The translation data consists of ``.po`` files, and they are managed in a separate
+repository from this code, ``https://github.com/creativecommons/cc-licenses-data``.
+This is typically checked out beside the ``cc-licenses`` repo, but can be put anywhere by changing the Django ``TRANSLATION_REPOSITORY_DIRECTORY`` setting.
```
Comment (Audiosutras) [inline README.rst:156]: I think we should just mention here that you can adjust the `TRANSLATION_REPOSITORY_DIRECTORY` setting by setting an environment variable of the same name.

File: README.rst [hunk 2]
```diff
@@ -132,3 +140,61 @@ Deployment with Dokku
...
+
+The .po files do *not* use the English messages as the message IDs, which would be
+the usual practice. This helps us to identify messages that should be translated
```
Comment (Audiosutras) [inline README.rst:159]: A bit confused by the wording of the second sentence.

File: cc_licenses/settings/base.py [hunk 3]
```diff
@@ -201,6 +201,9 @@
 # Percent translated that languages should be at or above
 TRANSLATION_THRESHOLD = 80
 
+# django-distill settings
+DISTILL_DIR = f"{ROOT_DIR}/build/"
+
```
Comment (Audiosutras) [inline cc_licenses/settings/base.py:206]:
```suggestion
```

File: cc_licenses/settings/base.py [hunk 4]
```diff
@@ -210,3 +213,9 @@
 # django-distill settings
 # Put the static files in the data repo under "build"
```
Comment (Audiosutras) [inline cc_licenses/settings/base.py:214]: I think we can remove this comment. We are not running `collectstatic` and all static files are currently coming from the vocab/bulma cdn.

File: licenses/models.py [hunk 5]
```diff
@@ -344,3 +416,58 @@ def sampling_plus(self):
...
+    branch_name = models.CharField(max_length=40)
+    legalcodes = models.ManyToManyField("LegalCode")
+    version = models.CharField(
+        max_length=3, help_text="E.g. '4.0'. Not required.", blank=True, default=""
+    )
+    language_code = models.CharField(
+        max_length=MAX_LANGUAGE_CODE_LENGTH,
+        help_text="E.g. 'en', 'en-ca', 'sr-Latn', or 'x-i18n'. Case-sensitive?",
+    )
+    last_transifex_update = models.DateTimeField(
```
Comment (Audiosutras) [inline licenses/models.py:461]: does this field need to be tz aware? Or will the transifex api return utc time?

Comment (Audiosutras) [reply licenses/models.py:461]: I'm just wondering in general what `time` we need to be tracking in the project and then displaying as a part of the translation status page. 

More of a question than a fix I'm requesting.

File: licenses/models.py [hunk 6]
```diff
@@ -166,6 +196,40 @@ def insert_after(after_this, what_to_insert):
...
+                entry.msgid = entry.msgstr
+                entry.msgstr = ""
+            return new_pofile
+        else:
+            english_pofile = self.get_english_pofile()
+            internal_key_to_english_message = {
+                entry.msgid: entry.msgstr for entry in english_pofile
+            }
+            for entry in new_pofile:
+                entry.msgid = internal_key_to_english_message[entry.msgid]
```
Comment (Audiosutras) [inline licenses/models.py:230]: I got this key error when running `python manage.py upload_license_messages`
```
KeyError: 's3a4_if_you_share_adapted_material'
```

File: README.rst (hunk 1 above)
Comment (dpoirier) [reply README.rst:156]: Will add.

File: README.rst (hunk 2 above)
Comment (dpoirier) [reply README.rst:159]: I don't remember why, actually - asking in Slack.

File: cc_licenses/settings/base.py (hunk 3 above)
Comment (dpoirier) [reply cc_licenses/settings/base.py:206]: Will do.

File: README.rst [hunk 7]
```diff
@@ -132,3 +140,61 @@ Deployment with Dokku
...
+Second note: the initial implementation focuses on the 4.0 by-*
+licenses. Others will be added as time allows.
+
+The translation data consists of ``.po`` files, and they are managed in a separate
+repository from this code, ``https://github.com/creativecommons/cc-licenses-data``.
+This is typically checked out beside the ``cc-licenses`` repo, but can be put anywhere by changing the Django ``TRANSLATION_REPOSITORY_DIRECTORY`` setting,
+or setting the ``TRANSLATION_REPOSITORY_DIRECTORY`` environment variable.
+
+The .po files do *not* use the English messages as the message IDs, which would be
+the usual practice. FIXME: FILL IN THE REASON HERE. OR USE THE ENGLISH MESSAGES
```
Comment (Audiosutras) [inline README.rst]: I think we got an answer from TR about this paragraph. English is the default internally; but anything user facing should be language agnostic as to the default. Can we re-word this paragraph and remove the `FIXME`

**Review by Audiosutras — APPROVED** [review]
💯

**Pull request merged**